src/main/java/com/hazelcast/azure/AzureClientHelper.java

```diff
@@ -41,10 +49,17 @@ private AzureClientHelper() {
      * @return ComputeManager a client to manage compute resources
      */
     public static ComputeManager getComputeManager(Map<String, Comparable> properties) {
-        ApplicationTokenCredentials atc = new ApplicationTokenCredentials(
-                (String) AzureProperties.getOrNull(CLIENT_ID, properties),
-                (String) AzureProperties.getOrNull(TENANT_ID, properties),
-                (String) AzureProperties.getOrNull(CLIENT_SECRET, properties), null);
+        AzureTokenCredentials atc;
+        String clientId = AzureProperties.getOrNull(CLIENT_ID, properties);
+        if (StringUtils.isEmpty(clientId)) {
```

Comment:
Consider `isBlank`? (I remember @dikeert being angry at MS for not applying `trim` to parameters. 😁)